pkg/ccl/backupccl/backup_test.go

```diff
@@ -3616,6 +3616,174 @@ func TestBackupRestoreSequence(t *testing.T) {
 	})
 }
 
+func TestBackupRestoreSequenceOwnership(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	const numAccounts = 1
+	_, _, origDB, dir, cleanupFn := BackupRestoreTestSetup(t, singleNode, numAccounts, InitNone)
+	defer cleanupFn()
+	args := base.TestServerArgs{ExternalIODir: dir}
+
+	backupLoc := LocalFoo + `/d`
+	backupLocD2 := LocalFoo + `/d2`
+	backupLocD3 := LocalFoo + `/d3`
+	// Setup for sequence ownership backup/restore tests in the same database.
+	origDB.Exec(t, `CREATE DATABASE d`)
+	origDB.Exec(t, `CREATE TABLE d.t(a int)`)
+	origDB.Exec(t, `CREATE SEQUENCE d.seq OWNED BY d.t.a`)
+	origDB.Exec(t, `BACKUP DATABASE d TO $1`, backupLoc)
```

Comment:
Could we also add a test where we restore all tables back into a different database on the same cluster: something like `CREATE DATABASE restore_db; RESTORE d.* FROM $1 WITH into_db='restore_db';` and check that the dependency still holds.